### PR TITLE
Implement basic OIDC auth

### LINKED ISSUE #74: Basic Agent Auth
**As a CBO agent, I should be able to sign into Connie using credentials provided by so I can access my dashboard.**

# Acceptance Criteria

- [ ] A CBO agent can use a web browser to sign into Connie using credentials provided to them or created by their CBO
- [ ] A CBO agent is redirected to the Connie dashboard upon successful sign-in
- [ ] A CBO agent is presented with an error message if they input incorrect credentials in the sign-in page
- [ ] A CBO agent is redirected to the Connie landing page upon sign-out

# Tasks

- [ ] Implement sign-in flow using [Okta Customer Identity Cloud](https://www.okta.com/customer-identity/)

# Additional Info

The assignees of this issue should work with @chrisberno to learn more about the existing and desired sign-in flow with Okta.

### PR DESCRIPTION
**Pull request by AlexanderTsarapkine**
Closes #74 

Added a very basic agent auth, after speaking with Chris, we decided there was no reason to proceed with SAML because of the lack of support and no real benefit.

Implemented simply OIDC authentication. Will speak with frontend team + rest of the team on how they want the sign page etc to work (using button on dashboard in the meantime). 

Can implement workforce identity + customer identity in future now that we use OIDC protocol

Edit / note:
I will also spend a bit of time looking into server side auth instead of client side auth because of possible security benefits

### CODE REVIEW

**Comment by lohpaul9** [comment]
Closing and re-opening to check for integration with deployment.

**Comment by lohpaul9** [comment]
Next is throwing an error in production because it requires a [secret in production](https://next-auth.js.org/errors#no_secret). Fix can be found [here](https://stackoverflow.com/questions/70712152/how-to-fix-no-secret-warning-thrown-by-next-auth).

**Comment by AlexanderTsarapkine** [comment]
> Next is throwing an error in production because it requires a [secret in production](https://next-auth.js.org/errors#no_secret). Fix can be found [here](https://stackoverflow.com/questions/70712152/how-to-fix-no-secret-warning-thrown-by-next-auth).

Added env instructions in readme

**Comment by AlexanderTsarapkine** [comment]
Changed to draft while working on final touch ups

**Comment by lohpaul9** [comment]
Closing and re-opening to check for integration with deployment.

File: README.md [hunk 1]
```diff
@@ -49,8 +49,18 @@ View the latest deployment branch [here](https://develop.d2r9j66448p933.amplifya
...
 Deployment build setting specifications:
-
 - Changed node version to 20 to be compatible with building next.js applications to fulfil requirement >=18.17.0
 - Changed container image to be Amazon linux 2023 to solve [GLIBC_2.28 not found](https://stackoverflow.com/questions/72921215/getting-glibc-2-28-not-found) error after upgrading node version.
 - Added npmrc file as a linker between pnpm and npm because amplify does some part of its installation using npm [here](https://docs.aws.amazon.com/amplify/latest/userguide/monorepo-configuration.html#turborepo-pnpm-monorepo-configuration).
 - Since repository is public, AWS amplify app has no linked IAM service roles for [security purposes](https://docs.aws.amazon.com/amplify/latest/userguide/pr-previews.html) to enable pull request previews.
+
+## Environment Setup
+
+### Environment Variables
```
Comment (lohpaul9) [inline README.md:59]: Suggest adding a short write-up on starting an okta app for authentication here

File: README.md [hunk 2]
```diff
@@ -49,8 +49,18 @@ View the latest deployment branch [here](https://develop.d2r9j66448p933.amplifya
...
+Create a .env file at the root of the repository with the following variables:
+NEXTAUTH_URL=http://localhost:3000
+OKTA_OAUTH2_CLIENT_ID={OKTA APPLICATION CLIENT ID}
+OKTA_OAUTH2_CLIENT_SECRET={OKTA APPLICATION SECRET}
+OKTA_OAUTH2_ISSUER=https://{YOUR OKTA ACCOUNT URL (click top right and it will appear below your email)}
+SECRET=Some long random string
+
```
Comment (lohpaul9) [inline README.md:66]: Got this to run in production: fix required was [due to not being loaded in the right env environment](https://docs.aws.amazon.com/amplify/latest/userguide/ssr-environment-variables.html)

**Review by lohpaul9 — APPROVED** [review]
Looks good to me, runs in prod too. Mild edits recommended in README but we can merge first.

**Pull request merged**